tests/integration/test_integration.py

```diff
@@ -377,27 +399,29 @@ async def test_deploy_all_the_things(
 
 
 @pytest.mark.abort_on_fail
-async def test_given_charm_deployed_then_status_blocked(ops_test: OpsTest, vault_idle: Task):
+async def test_given_charm_deployed_then_status_blocked(
+    ops_test: OpsTest, vault_idle_blocked: Task
+):
     assert ops_test.model
-    await vault_idle
+    await vault_idle_blocked
 
     vault_app = get_app(ops_test.model)
     assert vault_app.status == "blocked"
 
 
 @pytest.mark.abort_on_fail
 async def test_given_certificates_provider_is_related_when_vault_status_checked_then_vault_returns_200_or_429(  # noqa: E501
-    ops_test: OpsTest, vault_idle: Task, self_signed_certificates_idle: Task
+    ops_test: OpsTest, vault_idle_blocked: Task, self_signed_certificates_idle: Task
```

Comment:
Does this test actually care if vault is blocked?

Reply:
No it doesn't care if the charm is blocked or active. However the charm shouldn't be in waiting for example.
`vault_idle` might be true while the leader is still waiting. We might talk to Vault before the snap is installed or started.